web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -173,6 +174,9 @@ public void setConnection(Connector connection) throws Exception {
             // TODO: applyPrediction("Plan");
         } catch (Exception e) {
             log.error(e.getMessage(), e);
+            if (this.logic instanceof BaseQueryLogic) {
+                this.getMetric().setPlan(((BaseQueryLogic) this.logic).getConfig().getQueryString());
```

Comment:
I would add a null check on the getMetric call

Reply:
Maybe not... given that we don't seem to be doing that anywhere else.  Seems error-prone though.

Reply:
Good suggestion, thanks.